### PR TITLE
SW-2719 Refactor DatePicker and utils to use luxon

### PR DESCRIPTION
**Pull request by karthikbtf**
- removed moment* usage
- refactored DatePicker to use luxon, updated function signatures
- updated stories

### CODE REVIEW

**Comment by karthikbtf** [comment]
publish rc

**Comment by karthikbtf** [comment]
publish rc

**Comment by karthikbtf** [comment]
publish rc

**Comment by karthikbtf** [comment]
publish rc

**Review by nickgraz — APPROVED** [review]
Nice work!

**Comment by karthikbtf** [comment]
> Nice work!

Thanks 🙏 I realize I can't change the onChange signature radically, adds risk in migration.
Going to make one more change that supports previous (deprecated) onChange signature and add a new function that supports luxon/DateTime - gives clients time to migrate to new function.

**Comment by karthikbtf** [comment]
publish rc

**Comment by karthikbtf** [comment]
publish rc

**Comment by karthikbtf** [comment]
publish rc

**Comment by karthikbtf** [comment]
publish rc

**Comment by karthikbtf** [comment]
publish rc

**Comment by karthikbtf** [comment]
publish rc

**Comment by karthikbtf** [comment]
publish rc

File: src/components/DatePicker/DatePicker.tsx [hunk 1]
```diff
@@ -1,47 +1,91 @@
...
+import { TextField, TextFieldProps } from '@mui/material';
 import { LocalizationProvider, DesktopDatePicker } from '@mui/x-date-pickers';
-import { AdapterMoment } from '@mui/x-date-pickers/AdapterMoment';
+import { AdapterLuxon } from '@mui/x-date-pickers/AdapterLuxon';
+import { DateTime, Settings } from 'luxon';
 import Icon from '../Icon/Icon';
 import './styles.scss';
+import { DateType, getDate, tz } from '../../utils/date';
+
+/**
```
Comment (nickgraz) [inline src/components/DatePicker/DatePicker.tsx:10]: Thanks for the note

**Review by nickgraz — APPROVED** [review]
Looks good, thanks for making it backward compatible! Have you already filed a ticket for "convert clients to use new date format"? If not I can make it

**Comment by nickgraz** [comment]
* **#505** <a href="https://app.graphite.dev/github/pr/terraware/web-components/505?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @karthikbtf and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by nickgraz** [comment]
@karthikbtf I just merged something in which caused conflicts in `package.json`, I pushed up the fix

**Comment by karthikbtf** [comment]
publish rc

> Looks good, thanks for making it backward compatible! Have you already filed a ticket for "convert clients to use new date format"? If not I can make it

I haven't yet, let me do that. Thanks 🙏

**Comment by karthikbtf** [comment]
publish rc

**Comment by karthikbtf** [comment]
publish rc

**Comment by karthikbtf** [comment]
publish rc

**Comment by karthikbtf** [comment]
> Looks good, thanks for making it backward compatible! Have you already filed a ticket for "convert clients to use new date format"? If not I can make it

I realized I had to make the existing `onChange` prop optional, otherwise clients will have to define both the old and new callbacks during migration.

**Pull request merged**